mpc-core/src/protocols/rep3.rs

```diff
@@ -349,10 +367,41 @@ impl<F: PrimeField, N: Rep3Network> PrimeFieldMpcProtocol<F> for Rep3Protocol<F,
         Ok(r * y_inv)
     }
 
+    fn inv_many(&mut self, a: &[Self::FieldShare]) -> IoResult<Vec<Self::FieldShare>> {
+        let r = (0..a.len())
+            .map(|_| self.rand())
+            .collect::<Result<Vec<_>, _>>()?;
+        let tmp = self.mul_many(a, &r)?;
+        let y = self.open_many(&tmp)?;
```

Comment:
do we also want a mul_open_vec for this to be more round-efficient

Reply:
ok mul_open_many does exist, use it here then

Reply:
fixed